src/GafferArnold/IECoreArnoldPreview/ShaderAlgo.cpp

```diff
@@ -125,6 +126,55 @@ std::vector<AtNode *> convert( const IECore::ObjectVector *shaderNetwork, const
 					continue;
 				}
 			}
+
+			// Check stringVectorData params to see if they are links to other shaders
+			// There is currently a bit of a problem, in that the one case we want to use this in
+			// is light filters, which are of type NODE, and are connected using AiNodeSetPtr/Array
+			// rather than AiNodeLinkOutput
+			//
+			// Currently just hardcoding to use AiNodeSetArray in the vector case.  This works for
+			// all use cases I'm aware of, but there's no reason you couldn't have a non-array
+			// connection of type NODE or an array connection of type RGB.
+			//
+			// Should we be somehow checking if the parameter type is NODE in both the StringData and
+			// StringVectorData case?  I don't see anything in the API to do this.
+			//
+			// Alternatively, should we use a token other than "link:" to connect in this way?
+			if( const StringVectorData *stringVectorData = runTimeCast<const StringVectorData>( pIt->second.get() ) )
+			{
+				const vector<string> &values = stringVectorData->readable();
+
+				std::vector< AtNode* > filters;
+				for( unsigned int i = 0; i < values.size(); i++ )
+				{
+					const string &value = values[i];
+					if( boost::starts_with( value, "link:" ) )
+					{
+						const string linkHandle = value.c_str() + 5;
+						ShaderMap::const_iterator shaderIt = shaderMap.find( linkHandle );
+						if( shaderIt != shaderMap.end() )
+						{
+							filters.push_back( shaderIt->second );
+						}
+						else
+						{
+							msg( Msg::Warning, "IECoreArnold::ShaderAlgo", boost::format( "Couldn't find shader handle \"%s\" for linking" ) % linkHandle );
+						}
+					}
+				}
+			
+				if( filters.size() )
+				{
+					AtArray *filterArray = AiArrayAllocate( filters.size(), 1, AI_TYPE_POINTER );
```

Comment:
I haven't always been consistent with this myself, but in the case that you already have a buffer (the vector), it's probably worth using a single call to `AiArrayConvert` to create the array. At the very least it's more readable, but I suspect for larger buffers it also has better performance.